org.eclipse.scanning.test/src/org/eclipse/scanning/test/command/PyExpresserTest.java

```diff
@@ -358,6 +385,58 @@ public void testDummyMalcolmModel() throws Exception {
 		assertEquals(expected, detector);
 	}
 
+	@Test
+	public void testSpiralModel() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)